test/libsolidity/syntaxTests/nameAndTypeResolution/510_large_storage_array_mapping.sol

```diff
@@ -2,4 +2,3 @@ contract C {
     mapping(uint => uint[2**100]) x;
 }
 // ----
-// Warning: (17-48): Variable covers a large part of storage and thus makes collisions likely. Either use mappings or dynamic arrays and allow their size to be increased only in small quantities per transaction.
```

Comment:
Will this error come back in the future? If not, this test serves no purpose.

Reply:
We should have this error.

Reply:
I think we should clearly specify what exactly should be an error, resp. warning, in which way. This variable does *not* occupy a lot of storage, unless it is filled. And we cannot count everything as if it was non-empty, because that's impossible for recursive structs.

Reply:
The idea behind this restriction is that `x[0][<crafted_number>]` and `x[1][<other_crafted_number>]` can point to the same slot and it is relatively easy to come up with the correct numbers. We should disallow such large arrays in storage in general.
We can stop counting at recursive structs, but we should descend into every possible path. The product of the sizes of the arrays should be bounded by some "small" number.

Actually, thinking about it: Whenever we compute a hash, the chain is broken, because it cannot be easily predicted anymore, and this is true for both mappings and dynamically-sized arrays.

Reply:
Not sure I'm contributing any thing meaningful to this discussion but from what I understand (and piecing together info from docs), for `x[0][crafted_number]` and `x[1][another_crafted_number]` to collide, `keccak256(0.keccak256(crafted_number)) == keccak256(1.keccak256(another_crafted_number))` and I suppose the likelihood of finding such a collision increases when each key can occupy several storage slots (`uint[2**100]` in this instance).

Reply:
The problem is that `slot(x[0][t]) - slot(x[0][0]) == t` because arrays are laid out contiguously, so you can compute the correct `t` to force a collision even without brute force. The only protection against this is making sure that large values for `t` are not possible so that the range of two arrays does not overlap.

Reply:
Taking a step back here: This change does not belong to this PR, does it?

Reply:
My takeaway from this review thread is that https://github.com/ethereum/solidity/pull/9187/files#r440891211 is against this  specific change i.e., warning must be reinstated and perhaps even an error depending on how severe we decide a potential collusion to be.